Lib/annotationlib.py

```diff
@@ -495,6 +496,8 @@ def call_annotate_function(annotate, format, *, owner=None, _is_evaluate=False):
     on the generated ForwardRef objects.
 
     """
+    if format == Format.VALUE_WITH_FAKE_GLOBALS:
+        raise ValueError("The VALUE_WITH_FAKE_GLOBALS format is for internal use only")
```

Comment:
I was surprised to see this exception.  I haven't given the matter a lot of thought, and I don't have a strongly held opinion yet.  Can you tell me why we should raise this exception here?  (My knee-jerk instinct: "special cases aren't special enough to break the rules.")

p.s. if we do keep these assertions, please drop `"The "`.  The text should read `"VALUES_WITH_FAKE_GLOBALS format is for internal use only"`.

Reply:
There is no meaningful value to return for `VALUE_WITH_FAKE_GLOBALS`; it's only useful within an annotate function.

Concretely, raising an exception here means that if a user-provided annotate function delegates to an API in annotationlib without further thought, it will raise an exception for VALUE_WITH_FAKE_GLOBALS, which is what we'd want.